spring-kafka/src/main/java/org/springframework/kafka/listener/IMessageListener.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.listener;
+
+/**
+ * Top level interface for listener types.
+ *
+ * @param <T> the type.
+ *
+ * @author Gary Russell
+ * @since 1.1
+ *
+ */
+public interface IMessageListener<T> {
```

Comment:
How about something like `Generic*` instead of `I*`?
I don't mind against [C#](http://www.tutorialspoint.com/csharp/csharp_interfaces.htm), but at the same time I don't want to have their structures in our public Java API.
Just IMHO :smile: